plugins/scaffolder-backend/src/scaffolder/tasks/NunjucksWorkflowRunner.ts

```diff
@@ -339,20 +333,16 @@ export class NunjucksWorkflowRunner implements WorkflowRunner {
 
           context.steps[step.id] = { output: stepOutput };
 
-          await task.emitLog(`Finished step ${step.name}`, {
-            stepId: step.id,
-            status: 'completed',
-          });
+          await stepTrack.markSuccessful();
         } catch (err) {
-          await task.emitLog(String(err.stack), {
-            stepId: step.id,
-            status: 'failed',
-          });
+          await taskTrack.markFailed(step, err);
+          stepTrack.markFailed();
```

Comment:
Just to keep things consistent I'm thinking it's best to just keep all of these async tbh, feeling that would reduce the number of surprises in the code.

Reply:
Sure makes sense